src/Stratis.Bitcoin.Features.Miner/BlockDefinition.cs

```diff
@@ -254,50 +267,46 @@ protected void OnBuild(ChainedHeader chainTip, Script scriptPubKey)
+        /// <summary>
+        /// Adds a transaction to the block from the given mempool entry.
+        /// </summary>
+        private void AddToBlock(TxMempoolEntry mempoolEntry)
         {
-            this.logger.LogTrace("({0}.{1}:'{2}')", nameof(iter), nameof(iter.TransactionHash), iter.TransactionHash);
+            this.logger.LogTrace("({0}.{1}:'{2}')", nameof(mempoolEntry), nameof(mempoolEntry.TransactionHash), mempoolEntry.TransactionHash);
 
-            this.block.AddTransaction(iter.Transaction);
+            this.block.AddTransaction(mempoolEntry.Transaction);
 
-            this.BlockTemplate.VTxFees.Add(iter.Fee);
-            this.BlockTemplate.TxSigOpsCost.Add(iter.SigOpCost);
+            this.BlockTemplate.VTxFees.Add(mempoolEntry.Fee);
+            this.BlockTemplate.TxSigOpsCost.Add(mempoolEntry.SigOpCost);
 
             if (this.NeedSizeAccounting)
-                this.BlockSize += iter.Transaction.GetSerializedSize();
+                this.BlockSize += mempoolEntry.Transaction.GetSerializedSize();
 
-            this.BlockWeight += iter.TxWeight;
+            this.BlockWeight += mempoolEntry.TxWeight;
             this.BlockTx++;
-            this.BlockSigOpsCost += iter.SigOpCost;
-            this.fees += iter.Fee;
-            this.inBlock.Add(iter);
-
-            //bool fPrintPriority = GetBoolArg("-printpriority", DEFAULT_PRINTPRIORITY);
-            //if (fPrintPriority)
-            //{
-            // LogPrintf("fee %s txid %s\n",
-            //  CFeeRate(iter->GetModifiedFee(), iter->GetTxSize()).ToString(),
-            //  iter->GetTx().GetHash().ToString());
-
-            //}
+            this.BlockSigOpsCost += mempoolEntry.SigOpCost;
+            this.fees += mempoolEntry.Fee;
+            this.inBlock.Add(mempoolEntry);
+
             this.logger.LogTrace("(-)");
         }
 
-        // Methods for how to add transactions to a block.
-        // Add transactions based on feerate including unconfirmed ancestors
-        // Increments nPackagesSelected / nDescendantsUpdated with corresponding
-        // statistics from the package selection (for logging statistics).
-        // This transaction selection algorithm orders the mempool based
-        // on feerate of a transaction including all unconfirmed ancestors.
-        // Since we don't remove transactions from the mempool as we select them
-        // for block inclusion, we need an alternate method of updating the feerate
-        // of a transaction with its not-yet-selected ancestors as we go.
-        // This is accomplished by walking the in-mempool descendants of selected
-        // transactions and storing a temporary modified state in mapModifiedTxs.
-        // Each time through the loop, we compare the best transaction in
-        // mapModifiedTxs with the next transaction in the mempool to decide what
-        // transaction package to work on next.
+        /// <summary>
+        /// Methods for how to add transactions to a block.
```

Comment:
"Method for.."

Reply:
@ferdeen I'll fix some of these but they are out of scope of this PR as this will still be refactored and/or possibly removed.